### PR TITLE
Lower deployment target and add availability checks to all Combine related code

### PR DESCRIPTION
**Pull request by RonKliffer**

### CODE REVIEW

**Comment by freak4pc** [comment]
Me and @RonKliffer had a discussion around this yesterday, after I've heard this use-case from several people.

There are people who actually use Combine in newer portions of their apps, e.g. iOS 11 apps that want to use this dependency with an availability clause. I think it's a valid use-case overall. WDYT @jasdev @jdisho ?

File: Sources/Operators/CombineLatestMany.swift [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 import Combine
 
+@available(iOS 13.0, *)
```
Comment (freak4pc) [inline Sources/Operators/CombineLatestMany.swift:11]: All tests should stay iOS 13 and up. Only the target itself should be iOS 11+ 
I'm actually not sure it's possible, but worth checking

**Comment by jdisho** [comment]
Seems valid case to me too.👌

**Comment by jdisho** [comment]
Should we do the same thing for SPM?

**Comment by jasdev** [comment]
> WDYT @jasdev @jdisho?

Cool with me—no concerns on my end as long as you also think this is the right call for 11.x+ folks. 👌🏽

**Pull request merged**